keyboards/projectd/75/iso/keymaps/default/keymap.c

```diff
@@ -0,0 +1,97 @@
+/* Copyright 2023 GSKY <gskyGit@gsky.com.tw>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+enum my_keycodes {
+    RGB_R = QK_USER,
+    RGB_G,
+    RGB_B,
+    RGB_W,
+    SW_cy,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+    switch (keycode) {
+
+        case RGB_R:
+            if (record->event.pressed) {
+                rgb_matrix_sethsv(0, 255, 255);
+                rgb_matrix_mode(1);
+            }
+            return false; /* Skip all further processing of this key */
+
+        case RGB_G:
+            if (record->event.pressed) {
+                rgb_matrix_sethsv(85, 255, 255);
+                rgb_matrix_mode(1);
+            }
+            return false; /* Skip all further processing of this key */
+
+        case RGB_B:
+            if (record->event.pressed) {
+                rgb_matrix_sethsv(170, 255, 255);
+                rgb_matrix_mode(1);
+            }
+            return false; /* Skip all further processing of this key */
+
+        case RGB_W:
+            if (record->event.pressed) {
+                rgb_matrix_sethsv(0, 0, 255);
+                rgb_matrix_mode(1);
+            }
+            return false; /* Skip all further processing of this key */
+
+        case SW_cy:
+            rgb_matrix_sethsv(0, 255, 255);
+            rgb_matrix_mode(13);
+            return false; /* Skip all further processing of this key */
+
+        
+        default:
+            return true; /* Process all other keycodes normally */
+    }
+};
```

Comment:
generally, we want the default/via keymaps to be prestine and simple.  Little to no custom code in them. 

That said, this could be moved to iso.c, and us `QK_KB` instead of `SAFE_RANGE`.  That would have the benefit of being accessible to anyone creating their own keymap, and would reduce the amount of duplicate/redundant code.